backend/telemetry_core/src/aggregator/inner_loop.rs

```diff
@@ -422,16 +425,19 @@ where
                     }
                 };
 
+                // TODO: untie serialization and updating data
                 let mut feed_message_serializer = FeedMessageSerializer::new();
                 self.node_state
                     .update_node(node_id, payload, &mut feed_message_serializer);
 
                 if let Some(chain) = self.node_state.get_chain_by_node_id(node_id) {
                     let genesis_hash = chain.genesis_hash();
-                    self.finalize_and_broadcast_to_chain_feeds(
-                        &genesis_hash,
-                        feed_message_serializer,
-                    );
+                    if self.send_node_data {
```

Comment:
Looks like we can move this condition one level up and skip some more work that way